app/src/main/java/mod/trindadedev/manage/theme/ThemeManager.java

```diff
@@ -0,0 +1,128 @@
+package mod.trindadedev.manage.theme;
+
+import android.content.Context;
+import android.view.ContextThemeWrapper;
+import android.content.SharedPreferences;
+
+import androidx.appcompat.app.AppCompatDelegate;
+
+import com.google.android.material.color.DynamicColors;
+
+import com.besome.sketch.SketchApplication;
+import com.sketchware.remod.R;
+
+public class ThemeManager {
+    
+    public static final String THEME_PREF = "themedata";
+    
+    public static final String THEME_KEY = "theme";
+    public static final String DYNAMIC_KEY = "dynamic_theme";
+    public static final String VIEW_PANE_KEY = "view_pane_theme";
+    
+    public static final int THEME_SYSTEM = 0;
+    public static final int THEME_LIGHT = 1;
+    public static final int THEME_DARK = 2;
+    
+    public static final boolean VIEW_PANE_MD3 = false;
+
+    // apply Sketchware theme (system, light, dark)
+    public static void applyTheme(Context context, int type) {
+        savePreference(context, THEME_KEY, type);
+
+        switch (type) {
+            case THEME_SYSTEM:
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
+                break;
+            case THEME_LIGHT:
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
+                break;
+            case THEME_DARK:
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
+                break;
+            default:
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
+        }
+    }
+     
+    // apply dynamic colors if possible and if enabled.
+    public static void applyTheme (Context context, SketchApplication app) {
+        SharedPreferences preferences = context.getSharedPreferences(THEME_PREF, Context.MODE_PRIVATE);
+        if (preferences.getBoolean(DYNAMIC_KEY, true)) {
+            DynamicColors.applyToActivitiesIfAvailable(app);
+        }
+    }
+    
+    // apply view pane theme (md2(0) md3(1))
+    public static void applyTheme (Context context, boolean type) {
+        savePreference(context, VIEW_PANE_KEY, type);
+    }
+    
+    // get sketchware theme (system, light, dark)
+    public static int getSketchwareTheme(Context context) {
+        return getPreference(THEME_KEY, THEME_SYSTEM, context);
+    }
+    
+    // get view pane theme (md2(0), md3(1)) 
+    public static boolean getViewPaneUseMd3(Context context) {
+        return getPreference(VIEW_PANE_KEY, VIEW_PANE_MD3, context);
+    }
+    
+    public static Context viewPaneContextWrapper (Context context) {
+        if (getViewPaneUseMd3(context)) {
+             return new ContextThemeWrapper(context, R.style.ViewEditorThemeMd3);
+        } else {
+             return new ContextThemeWrapper(context, R.style.ViewEditorThemeMd2);
+        }
+    }
```

Comment:
Explain why we need this?

Reply:
to get the viewPane's theme (ContextThemeWrapper)

Reply:
I think it's useless feature.

Reply:
> I think it's useless feature. Sketchware by default creates MD2 project.

A